### DISCUSSION TITLE
Override `vitest.config.js > test.include` with CLI

### DISCUSSION THREAD

**Discussion by brillout** [vitest-dev/vitest#5348] [Q&A] [answered]
Is there a way to use the CLI to override the `test.include` setting set by `vitest.config.js`?

I've two kinds of tests: unit tests and E2E tests. By default `$ vitest` would run the unit tests.

In other words:

```js
export default {
  test: {
    // Unit tests
    include: ['src/**/*.spec.ts']
  }
}
```

While being able to `$ vitest --test.include 'examples/**/*.spec.ts'` (E2E tests). But `--test.include` doesn't exist.

**Comment by AriPerkkio** [comment]
Does this work: `$ vitest examples/**/*.spec.ts`?

**Reply by brillout** [reply]
It doesn't seem work: Vitest complains that no flles were found.

```
$ pnpm exec vitest examples/**/*.spec.ts

filter:  examples/preload/dev.spec.ts, examples/preload/prod.spec.ts
include: src/**/*.spec.ts
exclude:  **/node_modules/**, **/dist/**, **/cypress/**, **/.{idea,git,cache,output,temp}/**, **/{karma,rollup,webpack,vite,vitest,jest,ava,babel,nyc,cypress,tsup,build,eslint,prettier}.config.*
watch exclude:  **/node_modules/**, **/dist/**

No test files found, exiting with code 1
```

**Reply by AriPerkkio** [reply]
Oh right, filter and `include` are different things indeed. This was also requested in https://github.com/vitest-dev/vitest/issues/4861 but closed. I think `--include` CLI option would be useful. Any thoughts @sheremet-va?

**Reply by sheremet-va** [reply]
What I've been thinking is actually renaming include because too many people are getting confused by it (because of `coverage.include`). Then we can introduce a new CLI flag for it. We can discuss it in a team meeting.

**Reply by sheremet-va** [reply, marked as answer]
So, we had a team meeting and decided that it's better to wait for more demand for this feature. We believe that it's better to use Vitest [workspace feature](https://vitest.dev/guide/workspace.html) to organize your config:

```js
import { defineWorkspace } from 'vitest/config'

export default defineWorkspace([
  {
    test: {
      include: ['tests/**/*.unit.test.ts'],
      name: 'unit',
      environment: 'happy-dom',
    }
  },
  {
    test: {
      include: ['tests/**/*.e2e.test.ts'],
      name: 'e2e',
      environment: 'node',
    }
  }
])
```

You can then run it as `vitest --project=e2e`.

This also makes the new VSCode extension more performant.
